typescript/sdk/src/hook/HyperlaneHookDeployer.ts

```diff
@@ -392,9 +393,15 @@ export class HyperlaneHookDeployer extends HyperlaneDeployer<
         },
         'Setting routing hooks',
       );
+      const estimatedGas = await routingHook.estimateGas.setHooks(
+        routingConfigs,
+      );
       return this.multiProvider.handleTx(
         chain,
-        routingHook.setHooks(routingConfigs, overrides),
+        routingHook.setHooks(routingConfigs, {
+          gasLimit: addBufferToGasLimit(estimatedGas),
```

Comment:
is all this stuff bc of infinity's gasless setup?

Reply:
not explicitly because of infinity, I've noticed it on other chains as well but never looked into the root cause til now